algosdk/testing/dryrun.py

```diff
@@ -610,6 +634,89 @@ def build_dryrun_request(
             round=rnd,
         )
 
+    @classmethod
+    def _get_run_mode(cls, app):
+        run_mode = "lsig"
+        if app is not None:
+            on_complete = (
+                app.get("on_complete")
+                if isinstance(app, dict)
+                else app.on_complete
+            )
+            run_mode = (
+                "clearp"
+                if on_complete == transaction.OnComplete.ClearStateOC
+                else "approv"
+            )
+        return run_mode
+
+    @classmethod
+    def _prepare_app(cls, app, sender):
+        if isinstance(app, dict):
+            app = App(**app)
+
+        if app.app_idx is None:
+            app.app_idx = 0
+
+        if app.accounts is not None:
+            accounts = []
+            for acc in app.accounts:
+                if isinstance(acc, str):
+                    accounts.append(
+                        Account(
+                            address=acc,
+                        )
+                    )
+                else:
+                    accounts.append(acc)
+            app.accounts = accounts
+
+        return app
+
+    @classmethod
+    def _prepare_lsig(cls, lsig, sender):
```

Comment:
@tzaffi Minor:  `sender` parameter is unused.  Is it by-design?  If so, can it be removed?

The remark applies to `_prepare_app` too.

Reply:
Closing the loop - Thanks for bringing in the feedback via https://github.com/algorand/py-algorand-sdk/pull/298/commits/3c75c626513800983607a022ee8db6e097f39c6b.